### PR TITLE
Use `dev-tool run bundle` in generated azure-sdk-for-js packages

### PR DESCRIPTION
**Pull request by jeremymeng**
for bunding.

- update test baselines.

#2059

### CODE REVIEW

**Review by deyaaeldeen — APPROVED** [review]
Thanks! we should also update the `files` list to not include the `rollup.config.js` file.

File: packages/autorest.typescript/test/integration/generated/appconfigurationexport/package.json [hunk 1]
```diff
@@ -18,17 +18,12 @@
...
-    "@rollup/plugin-node-resolve": "^13.1.3",
     "mkdirp": "^2.1.2",
-    "rollup": "^2.66.1",
-    "rollup-plugin-sourcemaps": "^0.6.3",
     "typescript": "~5.0.0",
     "uglify-js": "^3.4.9",
     "rimraf": "^5.0.0",
-    "dotenv": "^16.0.0"
+    "dotenv": "^16.0.0",
+    "@azure/dev-tool": "^1.0.0"
```
[the remark this answers is not in the document]
Comment (jeremymeng) [reply packages/autorest.typescript/test/integration/generated/appconfigurationexport/package.json:26]: yes it's only added when `azureSdkForJs` is true https://github.com/Azure/autorest.typescript/pull/2060/files/d9cba5b1c8aa5729ae34f84c7b8115d9018150ae#diff-c0268d76bd99f6fb8f8cdb15c036f8df312c03e7adfb022f56bad59beb5d68f0R186

**Comment by jeremymeng** [comment]
The smoke tests are failing because they do npm install and npm build, however, installation fails to get `@azure/dev-tool`.  Any suggestion to fix these? These tests have the AzureSdkForJs option.

**Comment by jeremymeng** [comment]
I changed smoke tests to not test for azure-sdk-for-js since dev-tool is not available outside of js-repo. Yeah it's unfortunate that we cannot test the scenario, but hopefully we can fix things quickly whenever generated packages failed to build in the js repo.

File: packages/autorest.typescript/src/generators/static/packageFileGenerator.ts [hunk 2]
```diff
@@ -189,6 +182,19 @@ function regularAutorestPackage(
...
+    packageInfo.scripts["build"] =
+      "npm run clean && tsc && dev-tool run bundle && npm run minify && mkdirp ./review && npm run extract-api";
+  } else {
+    packageInfo.devDependencies["@rollup/plugin-commonjs"] = "^24.0.0";
+    packageInfo.devDependencies["@rollup/plugin-json"] = "^6.0.0";
+    packageInfo.devDependencies["@rollup/plugin-multi-entry"] = "^6.0.0";
+    packageInfo.devDependencies["@rollup/plugin-node-resolve"] = "^13.1.3";
+    packageInfo.devDependencies["rollup"] = "^2.66.1";
+    packageInfo.devDependencies["rollup-plugin-sourcemaps"] = "^0.6.3";
+  }
```
Comment (MaryGao) [inline packages/autorest.typescript/src/generators/static/packageFileGenerator.ts:196]: Thanks for contributing this! Could you help understand below questions?

- Does this apply for both RLC, HLC and Modular?
- Doest this change apply for both swagger and typespec path?

If yes I think we still have changes in rlc-common and typespec-ts.

File: packages/autorest.typescript/test/integration/generated/appconfiguration/package.json [hunk 3]
```diff
@@ -18,17 +18,12 @@
...
-    "@rollup/plugin-node-resolve": "^13.1.3",
     "mkdirp": "^2.1.2",
-    "rollup": "^2.66.1",
-    "rollup-plugin-sourcemaps": "^0.6.3",
     "typescript": "~5.0.0",
     "uglify-js": "^3.4.9",
     "rimraf": "^5.0.0",
-    "dotenv": "^16.0.0"
+    "dotenv": "^16.0.0",
+    "@azure/dev-tool": "^1.0.0"
```
Comment (MaryGao) [inline packages/autorest.typescript/test/integration/generated/appconfiguration/package.json:26]: Could you help understand why this change happens in `not azureSDKForJs` env?

https://github.com/Azure/autorest.typescript/blob/main/packages/autorest.typescript/test/commands/test-swagger-gen.ts#L35-L45

File: packages/autorest.typescript/test/smoke/generated/arm-package-deploymentscripts-2019-10-preview/package.json [hunk 4]
```diff
@@ -21,17 +21,17 @@
...
     "@rollup/plugin-multi-entry": "^6.0.0",
     "@rollup/plugin-node-resolve": "^13.1.3",
-    "mkdirp": "^2.1.2",
     "rollup": "^2.66.1",
-    "rollup-plugin-sourcemaps": "^0.6.3",
-    "typescript": "~5.0.0",
-    "uglify-js": "^3.4.9",
-    "rimraf": "^5.0.0",
-    "dotenv": "^16.0.0"
+    "rollup-plugin-sourcemaps": "^0.6.3"
```
Comment (MaryGao) [inline packages/autorest.typescript/test/smoke/generated/arm-package-deploymentscripts-2019-10-preview/package.json:34]: > The smoke tests are failing because they do npm install and npm build, however, installation fails to get @azure/dev-tool. Any suggestion to fix these? These tests have the AzureSdkForJs option.

Is the issue existing? I assume in non js-repo there will have no real change just order change.

File: packages/autorest.typescript/test/smoke/generated/arm-package-deploymentscripts-2019-10-preview/package.json [hunk 5]
```diff
@@ -50,7 +50,6 @@
     "src/**/*.ts",
     "README.md",
     "LICENSE",
-    "rollup.config.js",
```
Comment (MaryGao) [inline packages/autorest.typescript/test/smoke/generated/arm-package-deploymentscripts-2019-10-preview/package.json:53]: Is the deletion of this line expected in non js-repo?

Comment (jeremymeng) [reply packages/autorest.typescript/test/smoke/generated/arm-package-deploymentscripts-2019-10-preview/package.json:53]: Yes, there's no value in packing rollup config file.  it's for dev environment only.

File: packages/autorest.typescript/src/generators/static/packageFileGenerator.ts (hunk 2 above)
Comment (jeremymeng) [reply packages/autorest.typescript/src/generators/static/packageFileGenerator.ts:196]: Yes you are right,  any packages in the js repo should use dev-tool to bundle.

File: packages/autorest.typescript/test/smoke/generated/arm-package-deploymentscripts-2019-10-preview/package.json (hunk 4 above)
Comment (jeremymeng) [reply packages/autorest.typescript/test/smoke/generated/arm-package-deploymentscripts-2019-10-preview/package.json:34]: Prior to this change it doesn't matter because all generated use `rollup` so works fine with npm install.  Now package generated for js repo would failed to install. the old `AzureSdkForJs` option (`--AzureSdkForJs=false`) really is NotAzureSdkForJs so I renamed it and added it for all smoke tests.

File: packages/autorest.typescript/src/generators/static/packageFileGenerator.ts (hunk 2 above)
Comment (jeremymeng) [reply packages/autorest.typescript/src/generators/static/packageFileGenerator.ts:196]: Turns out that you and Qiaoqiao already use `dev-tool run bundle` for js repo in rlc-common and typespec-ts :)

**Comment by jeremymeng** [comment]
@MaryGao @qiaozha any other feedback?

File: packages/autorest.typescript/test/integration/generated/appconfiguration/package.json [hunk 6]
```diff
@@ -47,14 +42,13 @@
...
     "LICENSE",
-    "rollup.config.js",
     "tsconfig.json",
     "review/*",
     "CHANGELOG.md",
     "types/*"
   ],
   "scripts": {
-    "build": "npm run clean && tsc && rollup -c 2>&1 && npm run minify && mkdirp ./review && npm run extract-api",
+    "build": "npm run clean && tsc && dev-tool run bundle && npm run minify && mkdirp ./review && npm run extract-api",
```
Comment (qiaozha) [inline packages/autorest.typescript/test/integration/generated/appconfiguration/package.json:51]: I am a little confused, this change is for azure-sdk-for-js: true  and here we are changing the integration test. right ? Has the integration test enabled azure-sdk-for-js: true ?

Comment (jeremymeng) [reply packages/autorest.typescript/test/integration/generated/appconfiguration/package.json:51]: From what I can find, this test doesn't specify azureSdkForJs option

https://github.com/Azure/autorest.typescript/blob/main/packages/autorest.typescript/test/commands/test-swagger-gen.ts#L628

By default we generate for JS repo

https://github.com/Azure/autorest.typescript/blob/c3d9f9bf3c5d45ff9fea680fe73918428e7721ed/packages/autorest.typescript/src/utils/autorestOptions.ts#L123

**Comment by jeremymeng** [comment]
What command do I need to run to update the failing RLC smoke tests? I did `smoke-test` locally but don't see any change afterwards

**Comment by qiaozha** [comment]
I think it's a known issue https://github.com/Azure/autorest.typescript/issues/1703, unfortunately, we don't have time to investigate this for now.
